Library/Homebrew/extend/os/linux/diagnostic.rb

```diff
@@ -0,0 +1,65 @@
+require "tempfile"
+
+module Homebrew
+  module Diagnostic
+    class Checks
+      alias generic_check_tmpdir_sticky_bit check_tmpdir_sticky_bit
+
+      def check_tmpdir_sticky_bit
+        message = generic_check_tmpdir_sticky_bit
+        return if message.nil?
+        message + <<~EOS
+          If you don't have administrative privileges on this machine,
+          create a directory and set the HOMEBREW_TEMP environment variable,
+          for example:
+            export HOMEBREW_TEMP=~/tmp
+            mkdir -p "$HOMEBREW_TEMP"
+            chmod 1755 "$HOMEBREW_TEMP"
+            echo "export HOMEBREW_TEMP=\"$HOMEBREW_TEMP\"" >> #{shell_profile}
+        EOS
+      end
+
+      def check_ld_vars
```

Comment:
I reckon this could exist in `diagnostic.rb` and check both `DYLD` and `LD` variables there. Thoughts?

Reply:
I suggested this structure. I'm just fine with a single `check_ld_vars` in generic `diagnostic.rb` that checks both if you're happy with that.

Reply:
Is the suggestion to check these variables unconditionally on both platforms?

Reply:
@maxim-belkin Yep 👍